### PR TITLE
Moving vis_contents changes onto a subsystem.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
## Description of changes
- Adds SSvis_contents_updates and appropriate procs.
- Changes existing vis_contents macros to be procs that use the subsystem.

## Why and what will this PR improve
vis_contents manipulation is expensive and current uses on /turf will benefit from being ordered during initialization.

## Authorship
Myself.

## Changelog
Nothing player-facing.

### CODE REVIEW

**Comment by MistakeNot4892** [comment]
Per Discord it's unclear if this is a good idea or not. It increased Crux init by 5s (85s -> 90s).
EDIT: It's unclear if this was due to vis_contents or not, more profiling might be needed.

File: code/__defines/subsystems.dm [hunk 1]
```diff
@@ -20,27 +20,28 @@
...
+#define SS_INIT_JOBS             11
+#define SS_INIT_CIRCUIT          10
+#define SS_INIT_GRAPH            9
+#define SS_INIT_OPEN_SPACE       8
+#define SS_INIT_ATOMS            7
+#define SS_INIT_PRE_CHAR_SETUP   6
+#define SS_INIT_CHAR_SETUP       5
+#define SS_INIT_MACHINES         4
+#define SS_INIT_ICON_UPDATE      3
+#define SS_INIT_VIS_CONTENTS     2
```
Comment (out-of-phaze) [inline code/__defines/subsystems.dm]: It should init after `SS_INIT_AIR` to avoid wasted updates before zones are built.

**Review by out-of-phaze — CHANGES_REQUESTED** [review]
That optimization commit I linked used SSmisc_late because it runs after SSair, which lets us skip updates that will be invalidated by zone building (a lot of them, surprisingly). I'd like to see it tested running after SSair rather than before.

File: code/__defines/subsystems.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/__defines/subsystems.dm]: Done.

**Review by out-of-phaze — APPROVED** [review]

**Pull request merged**